roms_tools/plot.py

```diff
@@ -474,65 +488,83 @@ def _add_boundary_to_ax(
         The projection for transforming coordinates.
 
     c : str, optional
-        Color of the grid boundary (default is 'red').
+        Color of the grid boundary (default is 'k').
     """
     proj = ccrs.PlateCarree()
 
-    # find corners
-    corners = [
-        (lon_deg[0, 0], lat_deg[0, 0]),
-        (lon_deg[0, -1], lat_deg[0, -1]),
-        (lon_deg[-1, -1], lat_deg[-1, -1]),
-        (lon_deg[-1, 0], lat_deg[-1, 0]),
+    xi_dim = [d for d in lon_deg.dims if d.startswith("xi_")][0]
+    eta_dim = [d for d in lon_deg.dims if d.startswith("eta_")][0]
+
+    edges = [
+        (
+            _get_edge(lon_deg, xi_dim, "start"),
+            _get_edge(lat_deg, xi_dim, "start"),
+            r"$\eta$",
+        ),  # left
+        (
+            _get_edge(lon_deg, xi_dim, "end"),
+            _get_edge(lat_deg, xi_dim, "end"),
+            r"$\eta$",
+        ),  # right
+        (
+            _get_edge(lon_deg, eta_dim, "start"),
+            _get_edge(lat_deg, eta_dim, "start"),
+            r"$\xi$",
+        ),  # bottom
+        (
+            _get_edge(lon_deg, eta_dim, "end"),
+            _get_edge(lat_deg, eta_dim, "end"),
+            r"$\xi$",
+        ),  # top
     ]
 
-    # transform coordinates to projected space
-    transformed_corners = [trans.transform_point(lo, la, proj) for lo, la in corners]
-    transformed_lons, transformed_lats = zip(*transformed_corners)
-
-    ax.plot(
-        list(transformed_lons) + [transformed_lons[0]],
-        list(transformed_lats) + [transformed_lats[0]],
-        "o-",
-        c=c,
-        label=label,
-    )
-
-    if with_dim_names:
-        for i in range(len(corners)):
-            if i in [0, 2]:
-                dim_name = r"$\xi$"
-            else:
-                dim_name = r"$\eta$"
-            # Define start and end points for each edge
-            start_lon, start_lat = transformed_corners[i]
-            end_lon, end_lat = transformed_corners[(i + 1) % len(corners)]
-
-            # Compute midpoint
-            mid_lon = (start_lon + end_lon) / 2
-            mid_lat = (start_lat + end_lat) / 2
-
-            # Compute vector direction for arrow
-            arrow_dx = (end_lon - start_lon) * 0.4  # Scale arrow size
-            arrow_dy = (end_lat - start_lat) * 0.4
-
-            # Reverse arrow direction for edges 2 and 3
-            if i in [2, 3]:
-                arrow_dx *= -1
-                arrow_dy *= -1
-
-            # Add arrow
+    for i, (lon, lat, dim_name) in enumerate(edges):
```

Comment:
Consider placing `if with_dim_names` outside the loop to avoid unnecessary iteration.

That may allow you to get rid of the conditional `ax.plot` call inside the loop and the use of `enumerate`